Bugfix :: Nullness :: Add warnings for instantiations of (T|null) which are not allowed in F# (frequent at Deserialize from S.T.Json) (#18057)
dotnet/fsharp: src/Compiler/FSComp.txt

tcPassingWithoutNullTononNullFunction,"You can remove this `nonNull` assertion."
3263,tcNullableToStringOverride,"With nullness checking enabled, overrides of .ToString() method must return a non-nullable string. You can handle potential nulls via the built-in string function."
3264,tcDowncastFromNullableToWithoutNull,"Nullness warning: Downcasting from '%s' into '%s' can introduce unexpected null values. Cast to '%s|null' instead or handle the null before downcasting."

> 3265,tcDisallowedNullableApplication,"Application of method '%s' attempted to create a disallowed nullable (T|null) type for '%s'. Such type will not report nulness warnings correctly."

Author: It also warns for structs, incl. struct anons and struct tuples.
Taking the change, just removing the "reference".

> 3265,tcDisallowedNullableApplication,"Application of method '%s' attempted to create a disallowed nullable (T|null) type for '%s'. Such type will not report nulness warnings correctly."

Author: Even usage for structs is reported, because such method will "sneak in" default values without being able to tell them apart.

> 3265,tcDisallowedNullableApplication,"Application of method '%s' attempted to create a nullable type ('T | null) for '%s'. Nullness warnings won't be reported correctly for such types."

Reviewer: Maybe make it more explicit why it's a problem.

```suggestion
3265,tcDisallowedNullableApplication,"Application of method '%s' attempted to create a nullable type ('T | null) for '%s', which does not support nullability. Nullness warnings won't be reported correctly for this type."
```

Reviewer 2: This might be confusing, since `System.Nullable` exists, and people think about it as well when they thing "nullabillity" in .NET.

Author: True.
I will go with the current message.

It is indeed likely that most current examples of this warning will come via structs.
